papaya-web/src/main/java/pl/edu/agh/papaya/rest/projects/service/SprintsRestService.java

```diff
@@ -81,60 +82,67 @@
         }
 
         @SuppressWarnings({"MultipleStringLiterals"})
-        Long timePlanned = AssertionUtil.requireNonNegative("timePlanned", sprintDto.getTimePlanned());
+        Long estimatedTimePlanned = AssertionUtil.requireNonNegative(
+                "estimatedTimePlanned", sprintDto.getEstimatedTimePlanned());
 
         Sprint created = sprintService.newSprint()
                 .withProject(project)
                 .withEnrollmentPeriod(enrollmentPeriod)
                 .withDurationPeriod(durationPeriod)
-                .withTimePlanned(Duration.ofMinutes(timePlanned))
+                .withEstimatedTimePlanned(Duration.ofMinutes(estimatedTimePlanned))
                 .create();
         return ResponseEntity.ok(sprintMapper.mapToApi(created));
     }
 
-    private ResponseEntity<SprintDto> closeSprint(Long sprintId, Duration timeBurned, Duration timePlanned) {
-        Sprint sprint = getValidSprint(sprintId);
-
-        if (sprint.getDateClosed() != null) {
-            throw new IllegalStateException("Sprint already closed");
-        }
-
-        LocalDateTime timeNow = LocalDateTime.now();
-
-        if (sprint.getSprintState(timeNow) != SprintState.FINISHED) {
-            throw new IllegalStateException("Sprint cannot be closed before it has ended");
-        }
-
-        Sprint changedSprint = sprintService.closeSprint(sprint, timeBurned, timePlanned, timeNow);
-        return ResponseEntity.ok(sprintMapper.mapToApi(changedSprint));
+    public boolean isInOverlapWithLastSprint(Project project, LocalDateTimePeriod durationPeriod) {
+        Optional<Sprint> lastSprintOpt = sprintService.getLastInProject(project.getId());
+        return lastSprintOpt.isPresent() && !durationPeriod.isAfter(lastSprintOpt.get().getDurationPeriod());
     }
 
+    @SuppressWarnings({"MethodLength"})
     public ResponseEntity<SprintSummaryDto> getSprintSummary(Long projectId, Long sprintId) {
```

Comment:
Na `MethodLength` bym jednak proponowała wydzielenie metod

Reply:
Na sam kod tej metody popatrzę po rebasie, bo tu się zmieniło parę rzeczy

Reply:
metoda nie jest długa dlatego że dużo robi tylko dlatego ze wymaga rozsądnego formatowania
nie będę rozdzielał metod jeśli nie ma to sensu tylko dlatego ze statyczna analiza placze ze jest o 1 linie za duzo

Reply:
Ja ogólnie bym proponował wyłączyć `MethodLength`, tym bardziej jeśli krzyczy o tak krótkie metody. Trudność w konserwacji i reużyciu metody powinna być mierzona przez jej złożoność cyklomatyczną a nie jej długość.

Reply:
Raczej już nie krzyczy, bo jest chyba od 25 linii - pewnie tylko suppress został na polu bitwy

Jak dla mnie ta zasada to jest spoko wyznacznik, że może warto przemyśleć wydzielenie metod, ale nie jakaś wyrocznia - zawsze można zwiększyć globalnie albo suppressować, jeśli jest powód, żeby metoda miała milion linijek

Reply:
Suppress potrzebny?